engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/execution/ReentrantLocking.scala

```diff
@@ -105,8 +84,27 @@ class ReentrantLocking(logger: TruffleLogger) extends Locking {
       "should already held write compilation lock during the operation"
     )
 
-  /** @inheritdoc */
-  override def acquireReadCompilationLock(): Long = {
+  def withWriteCompilationLock[T](where: Class[_], callable: Callable[T]): T = {
+    var lockTimestamp: Long = 0
```

Comment:
I do remember we talk about lock ordering in the past. Here we should check that _read compilation lock_ isn't held. But looks like that is done in `acquireWriteCompilationLock`.